ST_initialization.c

```diff
@@ -0,0 +1,301 @@
+static void _endInitialization(void);
+static void _saveAsInitializationConditions(void);
+
+/***************************** Externed variables **********************************/
+/* Note that in an ideal world we wouldn't need to extern any variables because 
+   every module would declate them in a header file. Hopefully we can get this
+   cleaned up soon! -CH */
+
+// these are SOILWAT variables that we need...
+extern SW_SOILWAT SW_Soilwat;
+extern SW_SITE SW_Site;
+extern SW_VEGPROD SW_VegProd;
+extern SW_WEATHER SW_Weather;
+
+extern pcg32_random_t grid_rng;         // Gridded mode's unique RNG.
+
+/* We need to seed these RNGs when using the gridded mode but do not use them in this file. */
+extern pcg32_random_t environs_rng;     // Used exclusively in ST_environs.c
+extern pcg32_random_t mortality_rng;    // Used exclusively in ST_mortality.c
+extern pcg32_random_t resgroups_rng;    // Used exclusively in ST_resgroups.c
+extern pcg32_random_t species_rng;      // Used exclusively in ST_species.c
+extern pcg32_random_t markov_rng;       // Used exclusively in SW_Markov.c
+
+extern Bool UseProgressBar;             // From ST_main.c
+extern Bool* _SomeKillage;              // From ST_mortality.c
+
+/********* Modular functions defined elsewhere ************/
+/* Again, we should clean this up eventually. -CH */
+
+void rgroup_Establish(void);
+void rgroup_PartResources(void);
+void rgroup_Grow(void); 
+void mort_Main(Bool* killed);
+void rgroup_IncrAges(void);
+void _kill_annuals(void);
+void _kill_maxage(void);
+void proportion_Recovery(void); 
+void _kill_extra_growth(void); 	
+void parm_Initialize(void);
+void Plot_Initialize(void);
+void copy_rgroup(const GroupType* src, GroupType* dest);
+void copy_species(const SpeciesType* src, SpeciesType* dest);
+
+/* Initializes the plot with whichever method you have specified with initializationMethod. 
+   This function takes care of EVERYTHING involved with initialization.
+   After calling this function you can load in the initialization information by calling 
+   loadInitializationConditions(). */
+void runInitialization(void){
+	_beginInitialization();
+
+    /* Dummy accumulators to ensure we do not collect statistics */
+	StatType *dummy_Dist, *dummy_Ppt, *dummy_Temp,
+  		*dummy_Grp, *dummy_Gsize, *dummy_Gpr, *dummy_Gmort, *dummy_Gestab,
+  		*dummy_Spp, *dummy_Indv, *dummy_Smort, *dummy_Sestab, *dummy_Sreceived;
+	FireStatsType *dummy_Gwf;
+
+	/* For iterating over gridCells */
+	int i, j;
+
+	/* For iterating over years */
+	IntS year;
+
+    /* Initialization is technically an iteration so we need to seed the RNGs. */
+    RandSeed(SuperGlobals.randseed, &environs_rng);
+    RandSeed(SuperGlobals.randseed, &mortality_rng);
+    RandSeed(SuperGlobals.randseed, &resgroups_rng);
+    RandSeed(SuperGlobals.randseed, &species_rng);
+    RandSeed(SuperGlobals.randseed, &grid_rng);
+    RandSeed(SuperGlobals.randseed, &markov_rng);
+
+    if (BmassFlags.yearly || MortFlags.yearly)
```

Comment:
@kpalmqui this was a good catch! I tried running gridded mode with bmassflags.yearly ON and the program crashed. parm_Initialize is a non-gridded function, and it won't even work unless the program changes directories and cycles through all cells.

It looks to me like we don't need the call to parm_Initialize at all. When we start gridded mode we use gridded mode functions to read all of the input files before doing anything else. Do you know what the point of rereading parameters between iterations was? It seems like Plot_Initialize and SXW_reset are sufficient for resetting the program between iterations. I tried removing the call to parm_Initialize and the program completed with expected output. In addition, @fdpierson noted that calling parm_Initialize multiple times could result in memory leaks.

This is an issue that must be addressed before moving forward. If you have any information on why parm_Initialize is called multiple times let me know, otherwise I believe we can remove it from gridded mode.

Reply:
I have opened issue #389 to further discuss this problem.

Reply:
We have determined that the call to parm_Initialize can be removed entirely.
Resolved by commit d192856.